### PR TITLE
Alt text and marking elements decorative 

### PR DESCRIPTION
**Pull request by s596757**
## References

#9682

## Code changes
added aria-hidden to code to prevent decorative images from being picked up by screen reader and also edited code to allow left and right side bars to be read by screen reader 

## User-facing changes

side bar icons are read by screen readers but decorative images are ignored. All decorative imagesa re no longer picked up by screen reader

## Backwards-incompatible changes

none

### CODE REVIEW

File: packages/launcher/src/widget.tsx [hunk 1]
```diff
@@ -174,8 +174,12 @@ export class Launcher extends VDomRenderer<ILauncher.IModel> {
                 icon={icon}
                 iconClass={classes(iconClass, 'jp-Icon-cover')}
                 stylesheet="launcherSection"
+                title={cat + ' Icon'}
+                aria-hidden={true}
               />
-              <h2 className="jp-Launcher-sectionTitle">{cat}</h2>
+              <h2 className="jp-Launcher-sectionTitle" title={cat + ' Title'}>
```
Comment (fcollonval) [inline packages/launcher/src/widget.tsx:180]: What is the reasoning to add title on the icon and on the header? At first glance it does not bring much value.

File: packages/ui-components/src/icon/labicon.tsx [hunk 2]
```diff
@@ -829,11 +829,20 @@ namespace Private {
...
     const titleNodes = svgNode.getElementsByTagName('title');
+    // make the title the aria label0
+
+    /// TODO set better unique id for title
     if (titleNodes.length) {
       titleNodes[0].textContent = title;
+      if (!titleNodes[0].id) {
+        titleNodes[0].id = performance.now().toString();
+      }
+      svgNode.setAttribute('aria-labelledby', titleNodes[0].id);
```
Comment (fcollonval) [inline packages/ui-components/src/icon/labicon.tsx:840]: Looking at https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title, the `title` seems enough to provide an accessible _description_ for a SVG:

image

Which use case would you like to cover by adding the additional `aria-labelledby`?

File: packages/launcher/src/widget.tsx (hunk 1 above)
Comment (s596757) [reply packages/launcher/src/widget.tsx:180]: reads out title in screenreader then

File: packages/ui-components/src/icon/labicon.tsx [hunk 3]
```diff
@@ -829,11 +829,20 @@ namespace Private {
   export function setTitleSvg(svgNode: HTMLElement, title: string): void {
```
Comment (gabalafou) [inline packages/ui-components/src/icon/labicon.tsx:829]: I think this whole function is maybe dead code.

When I search the codebase, I can find only one place that it's called, inside the protected `_initSvg` method. And when I search the codebase for "_initSvg", I find exactly one call site, in the `svgElement` getter, and that call site does not pass `title`. So it looks to me that `setTitleSvg` isn't getting called anywhere in the JupyterLab codebase. Maybe there's some extension out there that subclasses `LabIcon` and redefines `_initSvg`, but not in this repo.

**Review by gabalafou — CHANGES_REQUESTED** [review]
It's really nice and exciting to see contributions to improve accessibility, thank you!

I have some doubts about these changes, though.

Before I weigh in, I should let you know that I do not use a screen reader regularly. In other words, I am a sighted developer who uses screen readers for development and testing.

I'm curious to know what operating system + screen reader combination you used to test the changes in this PR. The reason I ask is because I tested this PR using Mac OS + Voice Over, as well as Windows + NVDA. My doubts about this PR are based on my testing with those two screen readers. But perhaps you tested with a different screen reader and got different results.

I will now share the transcripts from my testing. First I will share the transcripts from NVDA (Windows) both before and after this PR. Then I will share the transcripts from Voice Over (Mac). All of the transcripts begin from the top of the region and stop at the level 2 heading labeled "Console." 

Transcript from NVDA before applying this PR:

```
heading level 2 Notebook
clickable Unlabeled graphic To get missing image descriptions, open the context menu
Python 3 (ipykernel)
heading level 2 Console
```

Transcript from NVDA after this PR applied:

```
Notebook Icon
heading level 2 Notebook
clickable Unlabeled graphic To get missing image descriptions, open the context menu
Python 3 (ipykernel)
Console Icon
heading level 2 Console
```

From my testing with NVDA, it seems that the main effect of this PR is to read out the icons. From my understanding, it is not helpful to read out icons for screen reader users (for example, refer to Sarah Soueidan's [Accessible Icon Buttons](https://www.sarasoueidan.com/blog/accessible-icon-buttons/#icon-sitting-next-to-text)).

I generated these transcripts in NVDA by using the "Say all" command (<kbd>NVDA</kbd> + <kbd>↓</kbd>). I also used the [object navigation shortcuts](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#ObjectNavigation), and got the same results.

Now let's look at Voice Over.

Here's the transcript from Voice Over when loading JupyterLab from the current main branch:

```
notebook content, region
image
heading level 2, Notebook
Python 3 (ipykernel), group
To get missing image descriptions, open the context menu, Unlabeled image
Python 3 (ipykernel), group
Python 3 (ipykernel)
image
heading level 2, Console
```

Transcript from Voice Over with this PR applied:

```
notebook content, region
Notebook Icon, group
image
heading level 2, Notebook
Python 3 (ipykernel), group
To get missing image descriptions, open the context menu, Unlabeled image
Python 3 (ipykernel), group
Python 3 (ipykernel)
Console icon, group
image
heading level 2, Console
```

As with NVDA, the main effect seems to be to read the icon, which I don't think is desired. 

My hunch is that we should probably undo the places where the title attribute has been added, but we should keep the place where aria-hidden is added. However, if I'm missing or misunderstanding anything, please let me know. 

If we decide to go down the route I suggest, we will need to modify the LabIcon class so that the aria-hidden attribute gets passed through to the rendered DOM.

Once again, thank you for your contribution!

File: packages/launcher/style/base.css
[the remark this answers is not in the document]
Comment (s596757) [reply packages/launcher/style/base.css:1]: I have removed the title attributes for the icons however I am unsure of how exactly you want me to amend the code to have labicon altered to pick up aria-labelledby by the dom and where you want this done. Any advice?

**Pull request closed without merging**